### ISSUE TITLE
avoid the need to lookup user DN inside the certificate

### ISSUE THREAD

**Issue by lecriste** [dmwm/CRABClient#4817]
As of OpenSSL 1.1.1, the default returned DN format is a comma-separated list.
Such format is not compatible with the previous slash-separated list and results in the client failing submission with
`Submitfail: Impossible to retrieve proxy`

More details:
https://hypernews.cern.ch/HyperNews/CMS/get/computing-tools/4916.html

**Comment by belforte** [comment]
We discussed with @mmascher the possibility to resolve this at the root by avoiding looking up the DN in the usercert.pem by using the CERN user name (aka crab_userhn) for the name of the credential in
myrpoxy instead of the has of the DN.
The relevant code in in WMCore
https://github.com/dmwm/WMCore/blob/42e2e17cdd28ee59876027ff77632aabb45360b4/src/python/WMCore/Credential/Proxy.py
@lecriste enquired with @amaltaro who confirmed that only CRAB is using that code, so we can look into changing it.
I am not rushing to do this since there could be operational complications if we change the credential name since the changes has to be done both client and server side, we need the server to be able to deal with both format during the transition.
Anyhow it looks like OpenSSL 1.1.1 will not be in use in Scientific Linux for a while, it was updated at DESY but other sites (starting with CERN) appear happy to stick with what is in the SL repository where RedHat will support OpenSSL 1.0.2 as long as SL7 is supported (privater communication from Maarten Litmaath).

**Comment by belforte** [comment]
Note that there is no need to use the VOMS group and role in the name of the credentil in myproxy since myproxy is unaware of proxy extensions and the stored credential can be used for any VOMS FQAN , as already done via:
https://github.com/dmwm/WMCore/blob/42e2e17cdd28ee59876027ff77632aabb45360b4/src/python/WMCore/Credential/Proxy.py#L611-L631

**Comment by belforte** [comment]
We checked with @amaltaro who confirmed that this part of WMCore/Credential is only used by CRAB, so can make a PR to remove the dependency on openssl and solve at the core. But that takes some time and given the operational complications described above, it is not clear that we'll ever do.

In the meanwhile we'll introduce a simple sanity check in the client so that if an incompatible OpenSSL version is used, it fails immediately. Will open a new issue.

**Comment by belforte** [comment]
this has become rather urgent because the login name for the credential in myproxy also includes the REST host name, and with myultile REST pointing to e.g. preprod db, it has become almost impossible to do thing consistently, currently submission must be sent to the same REST host as inidcated in TW config. which makes not sense given CRAB architecture.

**Comment by belforte** [comment]
THE PLAN:
- [ ] 1. modify WMCore Proxy.py to support both kind of credential names: the hash and username_CRAB
- [ ] 2. modify CRABClient to create and keep up to date both credentials
- [ ] 3. modify TW to use only the new credential username_CRAB 
- [ ] 4. check/modify Crabserver REST to use only the new credential username_CRAB
- [ ] 5. check/modify ASO/Publisher ...whatever as well
- [ ] 6. when running smoothly cleanup CRABClient from old code and finally remove lookup of DN inside certificate

**Comment by belforte** [comment]
1. is now in my repo via: https://github.com/belforte/WMCore/commit/536fad80b44385170e9905569c0057488ebd93db

2. is in my repo now, via: https://github.com/belforte/CRABClient/commit/2f30b2499847abefcf8cc23c22331ade976243c3

and now:
```
belforte@lxplus774/CRABClient> myproxy-info -l belforte_CRAB
username: belforte_CRAB
owner: /DC=org/DC=terena/DC=tcs/C=IT/O=Istituto Nazionale di Fisica Nucleare/CN=Stefano Belforte belforte@infn.it
  renewal policy: /DC=ch/DC=cern/OU=computers/CN=vocms(052|058|030|0118|0105|0109).cern.ch|/DC=ch/DC=cern/OU=computers/CN=crab-(preprod|prod)-tw(02|01).cern.ch
  trusted retrieval policy: /DC=ch/DC=cern/OU=computers/CN=vocms(052|058|030|0118|0105|0109).cern.ch|/DC=ch/DC=cern/OU=computers/CN=crab-(preprod|prod)-tw(02|01).cern.ch
  timeleft: 719:34:48  (30.0 days)
belforte@lxplus774/CRABClient> 
```

**Comment by belforte** [comment]
2. is now in CRABClient repository HEAD: https://github.com/dmwm/CRABClient/pull/4887

**Comment by belforte** [comment]
1. PullRequest made   https://github.com/dmwm/WMCore/pull/9548
 until it is merged and tagged the CRABClient head needs to be used with my WMCore branch
```
git clone https://github.com/belforte/WMCore.git
cd WMCore
git checkout simplifyMyProxy-fix-9544
```

**Comment by belforte** [comment]
1. is now in WMCore 1.3.1.pre2

**Comment by belforte** [comment]
2. is not in `crab-prod` in CVMFS/CMSSW distribution. Need to start switching users, anyhow there's no action on client code anymore.

**Comment by belforte** [comment]
as of crab client and server 3.3.2004.rc3 all is working with new simpler proxy.
Only thing left is to cleanup old code someties in June/July

**Comment by belforte** [comment]
Now all TaskWorkers are using usermame_CRAB
No point in creating old-fashioned credentials anymore.
can cleanup code and check last box in https://github.com/dmwm/CRABClient/issues/4817#issuecomment-589841490

**Comment by belforte** [comment]
solved via #4922

**Comment by belforte** [comment]
solved via #4922
